### ISSUE TITLE
Incompatibility of Nanoc 4.12.7 with Haml 6.0.1 ?

### ISSUE THREAD

**Issue by luc-j-bourhis** [nanoc/nanoc#1595]
### Steps to reproduce

1. I created a brand new site with `nanoc create-site`
2. Then I translated the layout `default.html` to HAML
3. Then compilation crashes

I have set up a repository with the complete site: https://github.com/luc-j-bourhis/nanoc-haml-crash-reproducer 

### Expected behavior

It should not crash

### Actual behavior

It does crash

```
ArgumentError: wrong number of arguments (given 2, expected 0..1)

Current item: /index.html (:default representation)

  0. /Users/luc/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/temple-0.8.2/lib/temple/engine.rb:44:in `initialize'
  1. /Users/luc/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/nanoc-4.12.7/lib/nanoc/filters/haml.rb:25:in `new'
  2. /Users/luc/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/nanoc-4.12.7/lib/nanoc/filters/haml.rb:25:in `run'
  ....
```

### Details

(Describe other details, if any, that you believe might be relevant.)

### Crash log

The crash log is committed in the repository mentioned above.

**Comment by luc-j-bourhis** [comment]
If I make the following change in my Gemfile

```
gem 'haml', '~> 5.2'
```

then the crash disappears. The problem is therefore indeed with Haml 6.

**Comment by doc75** [comment]
Is this a bug to be reported to Haml or is it something that could be fixed by Nanoc ?

**Comment by doc75** [comment]
Actually it is under work with #1593

**Comment by denisdefreyne** [comment]
The newest Haml version (v6) has a quite different API, and I have not been able to get it to work with Nanoc yet. I suggest sticking to Haml 5 for now. You can do so by specifying a version constraint in your project’s Gemfile:

```
gem 'haml', '~> 5.0'
```

**Comment by denisdefreyne** [comment]
This, for the most part, is fixed in #1593. The only thing I could not get to work (yet) is the `filtering` helper inside Haml, but perhaps (or hopefully) that is not used so much (as Haml provides its own filtering support).

**Comment by denisdefreyne** [comment]
I’ve updated the [documentation for the `filtering` helper](https://nanoc.app/doc/reference/helpers/#filtering) to mention that it’s compatible with ERB only.

Haml has built-in filtering support, and I imagine it makes more sense to use that, rather than Nanoc’s support.

With this, I think this issue can now be closed!
